### PR TITLE
SSE-3074 : Added tests related to register journey

### PR DESCRIPTION
**Pull request by raghu-kora**
Added the missing tests related to registration journey

### CODE REVIEW

File: acceptance-tests/features/register.feature [hunk 1]
```diff
@@ -18,17 +18,20 @@ Feature: Users can sign up to the self-service experience
...
     Scenario Outline: User can register allowed email addresses
-
       When they submit the email <email>
       Then they should be redirected to the "/register/enter-email-code" page
 
       Examples:
         | email                                     |
-        | "test-user@test.gov.uk" |
-        | "test-user@digital.cabinet-office.gov.uk"      |
+        | "test-user@test.gov.uk"                   |
```
Comment (MartinH-DI) [inline acceptance-tests/features/register.feature:26]: See comment below re allowed test domains

File: express/resources/allowed-test-domains.txt
Comment (MartinH-DI) [inline express/resources/allowed-test-domains.txt:1]: It was agreed (back in Karol's time) that we should restrict the allowed test domains to within test.gov.uk.

Comment (raghu-kora) [reply express/resources/allowed-test-domains.txt:1]: once we have capability to run these tests on full-stack we have to make sure about the domains we support 
I,e the reason I have added all the supported domains

File: acceptance-tests/features/register.feature (hunk 1 above)
Comment (raghu-kora) [reply acceptance-tests/features/register.feature:26]: as mentioned below to cover all the supported ones I have added the supported list

File: express/resources/allowed-test-domains.txt
Comment (BorisP-GDS) [reply express/resources/allowed-test-domains.txt:1]: The only domains we need here are :
digital.cabinet-office.gov.uk
test.gov.uk

**Review by BorisP-GDS — CHANGES_REQUESTED** [review]
The only domains we need here are :
digital.cabinet-office.gov.uk
test.gov.uk

File: acceptance-tests/features/register.feature [hunk 2]
```diff
@@ -37,6 +40,16 @@ Feature: Users can sign up to the self-service experience
...
         | "the-config-file-is-broken-add-a-dot-before-gov.uk@sneakygov.uk" |
 
+  Rule: User validating the links in registration page
+    Scenario Outline:User validating links in enter-email address page
+      Given the user is on the "/register" page
+      When they click on the '<linkName>' link
+      Then they should be directed to the URL '<address>'
+      Examples:
+        | linkName       | address                                                 |
+        | Contact us     | https://www.sign-in.service.gov.uk/contact-us?adminTool |
```
Comment (BorisP-GDS) [inline acceptance-tests/features/register.feature:50]: This test should be domain independent or domain dynamic.
Verify that link ends in "/contact-us" or make https://www.sign-in.service.gov.uk dynamic depending on environment.

File: express/resources/allowed-test-domains.txt
Comment (raghu-kora) [reply express/resources/allowed-test-domains.txt:1]: please let me know if there any specific reason I have to be aware of restricting the tests to only these 2 domains 
then how we can able to make sure about the supported domains without having the tests

Comment (BorisP-GDS) [reply express/resources/allowed-test-domains.txt:1]: The reason is that we are testing our domain restriction functionality - not the domain list itself (whatever is in the allowed list goes through, it does not matter what is in the list), and for that purpose, this list is optimal.
The domain list is dynamic, so you would have to write new test every time domain is added or removed, which is unnecessary, as long as we are testing that whatever is on the list goes through.

File: acceptance-tests/features/register.feature (hunk 2 above)
Comment (raghu-kora) [reply acceptance-tests/features/register.feature]: updated in the tests

**Review by MartinH-DI — APPROVED** [review]
Looks good

**Review by BorisP-GDS — APPROVED** [review]

**Pull request merged**